test_downstream/test_dsurf/test_Surface.py

```diff
@@ -14,9 +15,31 @@ def test_Surface(algo: types.ModuleType, S: int) -> None:
     assert [*surface] == [None] * S
     assert [*surface.lookup()] == [None] * S
 
-    for T in range(min(2**S, 100)):
+    for T in range(100):
         site = surface.ingest(T)
         if site is not None:
             assert surface[site] == T
         assert [*surface] == [*surface.lookup()]
         assert [*zip(surface.lookup(), surface)] == [*surface.enumerate()]
+
+
+@pytest.mark.parametrize("algo", [steady_algo, stretched_algo, tilted_algo])
```

Comment:
I would test by doing updates to one surface by stepping manually and the other by stepping many (for 3 or 4 manually chosen step sizes, one of them very large); go ahead and just store the negative ingest time as the data item (ie 0 -1 -2 etc)


also should probably test the case where num_ingests is 0